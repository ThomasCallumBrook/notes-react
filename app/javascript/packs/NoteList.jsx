import React from 'react'
import NoteSummary from './NoteSummary'


class NoteList extends React.Component{
  constructor(props){
    super(props)
  }
  render(){
    return(
        <div id="note-list"   >
          {this.props.notes.map((note) => {
              return (
                <NoteSummary {...note} key={note.id}
                  deleteNote={this.props.deleteNote}
                  currentNote={this.props.currentNote}
                />
              )
            })
          }
        </div>
    )
  }
}
export default NoteList;
