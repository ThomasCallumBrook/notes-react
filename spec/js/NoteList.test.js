import React from 'react';
import NoteList from 'NoteList';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import sinon from 'sinon';
import toJson from 'enzyme-to-json';

const notes = [
  {id: 1, title: "Note A", content: "What a wonderful React Monday", tags: "learning"},
  {id: 2, title: "Note B", content: "What a wonderful React Tuesday", tags: "learning"},
]
test('NoteList should render as expected', () => {
  const component = shallow(<NoteList notes={notes}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
