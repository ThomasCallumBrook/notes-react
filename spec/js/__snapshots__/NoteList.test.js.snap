// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList should render as expected 1`] = `
<div
  id="note-list"
>
  <NoteSummary
    content="What a wonderful React Monday"
    id={1}
    key="1"
    tags="learning"
    title="Note A"
  />
  <NoteSummary
    content="What a wonderful React Tuesday"
    id={2}
    key="2"
    tags="learning"
    title="Note B"
  />
</div>
`;
